src/resolvers-cassandra/Facts/queries.js

```diff
@@ -1,17 +1,81 @@
 'use strict';
 
+const Promise = require('promise');
+const cassandraConnector = require('../../clients/cassandra/CassandraConnector');
+
+/**
+ * @typedef {id: string, language: string, title: string, tags: string[], date: string, link: string, text: string, sources: string[]} Fact
+ */
+
+function cassandraRowToFact(row) {
+  return {
+    id: row.id,
+    language: row.eventlangcode,
+    title: row.title,
+    tags: row.detectedkeywords,
+    date: row.event_time,
+    link: row.sourceurl,
+    text: row.messagebody,
+    sources: row.pipeline && [row.pipeline]
+  };
+}
+
+function appendDefaultFilters(query, params) {
+  query += ' AND pipeline IN ("TadaWeb", "Bing", "CustomEvent")';
+  return {query: query, params: params};
+}
+
+function makeListQuery(args) {
+  const tagsCondition = args.tagFilter.map(_ => 'detectedkeywords CONTAINS ?').join(' OR '); // eslint-disable-line no-unused-vars
+  const query = `SELECT * FROM fortis.events WHERE (${tagsCondition})`;
+  const params = args.tagFilter.slice();
+  return appendDefaultFilters(query, params);
+}
+
 /**
  * @param {{pageSize: number, skip: number, tagFilter: string[]}} args
- * @returns {Promise.<{runTime: string, type: string, facts: Array<{id: string, language: string, title: string, tags: string[], date: string, link: string, text: string, sources: string[]}>}>}
+ * @returns {Promise.<{runTime: string, type: string, facts: Fact[]}>}
  */
 function list(args, res) { // eslint-disable-line no-unused-vars
+  return new Promise((resolve, reject) => {
+    if (!args || !args.tagFilter || !args.tagFilter.length) return reject('No tags specified to fetch');
+
+    const query = makeListQuery(args);
+    return cassandraConnector.executeQuery(query.query, query.params)
+    .then(rows => {
+      const facts = rows.map(cassandraRowToFact);
+
+      resolve({
+        facts: facts
```

Comment:
This is a cross-cutting concern that we should implement once for all the resolvers (e.g. as a decorator).